ts/nni_manager/rest_server/index.ts

```diff
@@ -0,0 +1,140 @@
+        // `shutdown()` is not a callback and should not be passed to NNIRestHandler.
+        app.use(this.urlPrefix, rootRouter(this.shutdown.bind(this)));
+        app.all('*', (_req: Request, res: Response) => { res.status(404).send(`Outside prefix "${this.urlPrefix}"`); });
+        this.server = app.listen(this.port);
+
+        const deferred = new Deferred<void>();
+        this.server.on('listening', () => {
+            this.logger.info('REST server started.');
+            deferred.resolve();
+        });
+        // FIXME: Use global handler. The event can be emitted after listening.
+        this.server.on('error', (error: Error) => {
+            this.logger.error('REST server error:', error);
+            deferred.reject(error);
+        });
+        return deferred.promise;
+    }
+
+    public shutdown(): Promise<void> {
+        this.logger.info('Stopping REST server.');
+        if (this.server === null) {
+            this.logger.warning('REST server is not running.');
+            return Promise.resolve();
+        }
+        const deferred = new Deferred<void>();
+        this.server.close(() => {
+            this.logger.info('REST server stopped.');
+            deferred.resolve();
+        });
+        // FIXME: Use global handler. It should be aware of shutting down event and swallow errors in this stage.
+        this.server.on('error', (error: Error) => {
+            this.logger.error('REST server error:', error);
+            deferred.resolve();
+        });
+        return deferred.promise;
+    }
+}
+
+/**
+ *  You will need to modify this function if you want to add a new module, for example, project management.
+ *
+ *  Each module should have a unique URL prefix and a "Router". Check express' reference about Router.
+ *  Note that the order of `use()` calls does mater and you must never put a router after 404 handler.
+ *  
+ *  In fact experiments management should have a separate prefix and module.
+ **/
+function rootRouter(stopCallback: () => Promise<void>): Router {
+    const router = Router();
+    router.use(bodyParser.json({ limit: '50mb' }));
+
+    // NNI Manager APIs
+    router.use('/api/v1/nni', createRestHandler(stopCallback));
+
+    // Download log files
+    router.use('/logs', express.static(getLogDir()));
+
+    // NAS architecture visualization
+    router.use('/netron', netronProxy());
+
+    // Web UI pages (put it last in case there is something strange in static directory)
+    router.use('/', express.static('static'));
+
+    // 404
+    router.all('*', (_req: Request, res: Response) => { res.status(404).send('Not Found'); });
+    return router;
+}
+
+function netronProxy(): Router {
+    const router = Router();
+    const proxy = httpProxy.createProxyServer();
+    router.all('*', (req: Request, res: Response): void => {
+        delete req.headers.host;
+        proxy.web(req, res, { changeOrigin: true, target: 'https://netron.app' });
```

Comment:
Seems that we don't have a test for this. Please make sure this change doesn't break netron.

Reply:
Now we do 🎉